From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ZHANG Yuntian <yt@radxa.com>
Date: Fri, 27 Jun 2025 07:54:52 +0000
Subject: [PATCH] HACK: fix compiler build error

We have no idea the size of the buffer, so we just clear what have been written.
---
 .../drm/sunxi_device/hardware/lowlevel_hdmi20/dw_dev.c      | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/drivers/video/drm/sunxi_device/hardware/lowlevel_hdmi20/dw_dev.c b/src/drivers/video/drm/sunxi_device/hardware/lowlevel_hdmi20/dw_dev.c
index a1f7766b033..42f19fda18e 100644
--- a/src/drivers/video/drm/sunxi_device/hardware/lowlevel_hdmi20/dw_dev.c
+++ b/src/drivers/video/drm/sunxi_device/hardware/lowlevel_hdmi20/dw_dev.c
@@ -262,15 +262,15 @@ ssize_t dw_hdmi_dump(char *buf)
 		i2cm->mode ? "fast" : "standard", i2cm->sfrClock / 100,
 		dw_i2cm_get_rate() / 10, dw_i2cm_get_rate() % 10);
 		printf("%s\n", buf);
-	memset(buf, 0x0, ARRAY_SIZE(buf));
+	memset(buf, 0x0, n);
 	n = 0;
 	n += dw_avp_dump(buf + n);
 	printf("%s\n", buf);
-	memset(buf, 0x0, ARRAY_SIZE(buf));
+	memset(buf, 0x0, n);
 	n = 0;
 	n += dw_phy_dump(buf + n);
 	printf("%s\n", buf);
-	memset(buf, 0x0, ARRAY_SIZE(buf));
+	memset(buf, 0x0, n);
 	n = 0;
 	return n;
 }
-- 
2.49.0

